src/lidar_tools/dsm_functions.py

```diff
@@ -375,6 +376,8 @@ def create_pdal_pipeline(
         Whether to filter low noise points, by default False.
     filter_high_noise : bool, optional
         Whether to filter high noise points, by default False.
+    hag_nn : float, optional
```

Comment:
I prefer long-form variables names (or maybe HAG is common enough?). maybe `height_above_ground_threshold`?